First micro-ROS app on Linux reviewed. (#183)
micro-ROS/micro-ROS.github.io: _docs/tutorials/core/first_application_linux/index.md

## Installing ROS 2 and the micro-ROS build system

First of all, install **ROS 2 Dashing Diademata** on your Ubuntu 18.04 LTS computer.

> To do so from binaries, via Debian packages, follow the instructions detailed

Reviewer: ```suggestion
To do so from binaries, via Debian packages, follow the instructions
```

> [here](https://index.ros.org/doc/ros2/Installation/Dashing/Linux-Install-Debians/) instead.

Reviewer: ```suggestion
from the [ROS Index](https://index.ros.org/doc/ros2/Installation/Dashing/Linux-Install-Debians/) website.
```

  the `fake_ping` correctly.

The diagram below clarifies the communication flow between these entities:

![pingpong](http://www.plantuml.com/plantuml/png/ZOwnIWGn48RxFCNFzSkoUG2vqce5jHEHi1dtWZkPa6GByNntavZY10yknMJu-ORlFwPiOjvvK-d3-M2YOR1uMKvHc93ZJafvoMML07d7h1NAE-DPWblg_na8vnwEx9OeZmzFOt1-BK7AzetJciPxCfRYVw1S0SbRLBEg1IpXPIvpUWLCmZpXIm6BS3addt7uQpu0ZQlxT1MK2r0g-7sfqbsbRrVfMrMwgbev3CDTlsqJGtJhATUmSMrMg5TKwaZUxfcttuMt7m00)

The app logic of this demo is contained in
[`app.c`](https://github.com/micro-ROS/micro-ROS-demos/blob/dashing/rcl/ping_pong/main.c).
A thorough review of this file is illustrative of how to create a micro-ROS publisher or subscriber.

Once the app has been created, the build step is in order.
Notice that, with respect to the four-steps workflow delined above, we would expect a configuration step to happen
before building the app. However, given that we are compiling micro-ROS in the host machine rather than in a board,
the cross-compilation implemented by this configure step is not required in this case.
We can then proceed to build the firmware and source the local installation:

```bash

> # Build step

Reviewer: ```suggestion
```
I propose to remove this comment given the explanation in previous paragraph.

ros2 run micro_ros_setup build_firmware.sh
source install/local_setup.bash
```

## Create the micro-ROS agent

The micro-ROS app is now ready to be connected to a micro-ROS agent to start talking with the rest of the ROS 2
world.

To do that, let's first of all create a micro-ROS agent:

```bash

> # Download micro-ROS-Agent packages

Reviewer: ```suggestion
```
Comment is explained in the previous paragraph already.

ros2 run micro_ros_demos_rcl ping_pong
```

## Testing the micro-ROS app

Now, we want to check that everything is working.

Open another command line. We are going to listen to the `ping` topic
with ROS 2 to check whether the micro-ROS Ping Pong node is correctly publishing the expected pings:

```bash

> source /opt/ros/$ROS_DISTRO/setup.bash

Reviewer: ```suggestion
source /opt/ros/dashing/setup.bash
```

... to be consistent with similar changes above.

ros2 topic echo /microROS/pong
```

And now, let's publish a `fake_ping` with ROS 2 from yet another command line:

```bash

> source /opt/ros/$ROS_DISTRO/setup.bash

Reviewer: ```suggestion
source /opt/ros/dashing/setup.bash
```

---
```

## Multiple Ping Pong nodes

One of the advantages of having an Linux micro-ROS app is that you don't need to buy a bunch of hardware in order to
test some multi-node micro-ROS apps.
So, with the same micro-ROS agent of the last section, let's open four different command lines and run the following on
each:

```bash
cd microros_ws

> source /opt/ros/$ROS_DISTRO/setup.bash

Reviewer: ```suggestion
source /opt/ros/dashing/setup.bash
```
